### ISSUE TITLE
Autocomplete with Partials

### ISSUE THREAD

**Issue by sjplatt** [ankane/searchkick#462]
Hi,

I'm struggling to figure out how to do autocomplete with partials. For example, when I type "skyrim", I want "The elder scrolls: skyrim" to show up.

Here is how the code is currently implemented:

```
class Game < ActiveRecord::Base
  searchkick autocomplete: ["name"]
  has_many :dlcs
  has_many :packages
end

  def autocomplete
    render json: Game.search(params[:query], autocomplete: true, limit: 10).map(&:name)
  end
```

I have tried other things suggested in other posts, but none of them seem to work. Thanks

**Comment by joshmn** [comment]
Do you want each result to show up in a partial, or do you want your autocomplete in its entirety to show up in a partial?

**Comment by sjplatt** [comment]
Hmm I think I didn't make this as clear as I could. I have a couple things I'm looking for and I'm not sure if the features are available:

Currently in my app when you start searching a list of suggestions appear below. Also a little grey version of the top suggestion appears in the search bar 

1) I want the suggestions to not only consider the start of the word (for example, if I type skyrim I want the elder scrolls skyrim to show up as a suggestion). Currently it only looks at the start of the word. I tried using operator: "or" but I got really strange suggestions. (This might be a stretch but I do have a method that can create a custom ordering of the names, not sure if this can be integrated. This is not necessary if the above can be done).
2) As you type a little grey "suggestion" appears in the search bar. Is it possible that when I press enter it will post that suggestion instead of the text the user types?
3) As I search, is it possible for the suggestions to ignore non alphanumeric characters. So if I type fallout new Vegas, fallout: new Vegas will appear as a suggestion.

I know this is a lot and thanks!
